[2024-11-24T10:07:37.091+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-11-24T10:07:37.260+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-11-24T10:07:28.068559+00:00 [queued]>
[2024-11-24T10:07:37.311+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-11-24T10:07:28.068559+00:00 [queued]>
[2024-11-24T10:07:37.312+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-11-24T10:07:38.024+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): sensor_data_consumer> on 2024-11-24 10:07:28.068559+00:00
[2024-11-24T10:07:38.076+0000] {standard_task_runner.py:64} INFO - Started process 921 to run task
[2024-11-24T10:07:38.112+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sensor_data_consumer', 'sensor_data_consumer', 'manual__2024-11-24T10:07:28.068559+00:00', '--job-id', '302', '--raw', '--subdir', 'DAGS_FOLDER/***_consumer.py', '--cfg-path', '/tmp/tmp2r1g0eog']
[2024-11-24T10:07:38.128+0000] {standard_task_runner.py:91} INFO - Job 302: Subtask sensor_data_consumer
[2024-11-24T10:07:38.482+0000] {task_command.py:426} INFO - Running <TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-11-24T10:07:28.068559+00:00 [running]> on host a9401edbac39
[2024-11-24T10:07:39.448+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Bich Ly' AIRFLOW_CTX_DAG_ID='sensor_data_consumer' AIRFLOW_CTX_TASK_ID='sensor_data_consumer' AIRFLOW_CTX_EXECUTION_DATE='2024-11-24T10:07:28.068559+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-24T10:07:28.068559+00:00'
[2024-11-24T10:07:39.455+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-11-24T10:07:39.965+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2024-11-24T10:07:40.035+0000] {spark_submit.py:344} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-streaming-kafka-0-10_2.12:3.4.2,org.apache.spark:spark-sql-kafka-0-10_2.12:3.4.2,org.apache.hadoop:hadoop-client:3.2.1 --name KafkaSparkHDFS /opt/***/dags/spark_streaming_job.py
[2024-11-24T10:08:16.987+0000] {spark_submit.py:495} INFO - :: loading settings :: url = jar:file:/home/***/.local/lib/python3.11/site-packages/pyspark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-11-24T10:08:19.244+0000] {spark_submit.py:495} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2024-11-24T10:08:19.248+0000] {spark_submit.py:495} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2024-11-24T10:08:19.257+0000] {spark_submit.py:495} INFO - org.apache.spark#spark-streaming-kafka-0-10_2.12 added as a dependency
[2024-11-24T10:08:19.262+0000] {spark_submit.py:495} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
[2024-11-24T10:08:19.263+0000] {spark_submit.py:495} INFO - org.apache.hadoop#hadoop-client added as a dependency
[2024-11-24T10:08:19.266+0000] {spark_submit.py:495} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-e2b88774-78c2-4e6a-9e81-5126471ab55e;1.0
[2024-11-24T10:08:19.266+0000] {spark_submit.py:495} INFO - confs: [default]
[2024-11-24T10:08:22.782+0000] {spark_submit.py:495} INFO - found org.apache.spark#spark-streaming-kafka-0-10_2.12;3.4.2 in central
[2024-11-24T10:08:23.355+0000] {spark_submit.py:495} INFO - found org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.4.2 in central
[2024-11-24T10:08:23.972+0000] {spark_submit.py:495} INFO - found org.apache.kafka#kafka-clients;3.3.2 in central
[2024-11-24T10:08:24.360+0000] {spark_submit.py:495} INFO - found org.lz4#lz4-java;1.8.0 in central
[2024-11-24T10:08:24.575+0000] {spark_submit.py:495} INFO - found org.xerial.snappy#snappy-java;1.1.10.3 in central
[2024-11-24T10:08:24.740+0000] {spark_submit.py:495} INFO - found org.slf4j#slf4j-api;2.0.6 in central
[2024-11-24T10:08:24.892+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-client-runtime;3.3.4 in central
[2024-11-24T10:08:25.280+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-client-api;3.3.4 in central
[2024-11-24T10:08:25.965+0000] {spark_submit.py:495} INFO - found commons-logging#commons-logging;1.1.3 in central
[2024-11-24T10:08:26.432+0000] {spark_submit.py:495} INFO - found com.google.code.findbugs#jsr305;3.0.0 in central
[2024-11-24T10:08:26.676+0000] {spark_submit.py:495} INFO - found org.apache.spark#spark-sql-kafka-0-10_2.12;3.4.2 in central
[2024-11-24T10:08:27.225+0000] {spark_submit.py:495} INFO - found org.apache.commons#commons-pool2;2.11.1 in central
[2024-11-24T10:08:27.884+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-client;3.2.1 in central
[2024-11-24T10:08:28.830+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-common;3.2.1 in central
[2024-11-24T10:08:29.185+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-annotations;3.2.1 in central
[2024-11-24T10:08:29.560+0000] {spark_submit.py:495} INFO - found com.google.guava#guava;27.0-jre in central
[2024-11-24T10:08:29.932+0000] {spark_submit.py:495} INFO - found com.google.guava#failureaccess;1.0 in central
[2024-11-24T10:08:30.040+0000] {spark_submit.py:495} INFO - found com.google.guava#listenablefuture;9999.0-empty-to-avoid-conflict-with-guava in central
[2024-11-24T10:08:30.258+0000] {spark_submit.py:495} INFO - found org.checkerframework#checker-qual;2.5.2 in central
[2024-11-24T10:08:30.355+0000] {spark_submit.py:495} INFO - found com.google.errorprone#error_prone_annotations;2.2.0 in central
[2024-11-24T10:08:30.465+0000] {spark_submit.py:495} INFO - found com.google.j2objc#j2objc-annotations;1.1 in central
[2024-11-24T10:08:30.590+0000] {spark_submit.py:495} INFO - found org.codehaus.mojo#animal-sniffer-annotations;1.17 in central
[2024-11-24T10:08:30.748+0000] {spark_submit.py:495} INFO - found commons-cli#commons-cli;1.2 in central
[2024-11-24T10:08:30.866+0000] {spark_submit.py:495} INFO - found org.apache.commons#commons-math3;3.1.1 in central
[2024-11-24T10:08:31.045+0000] {spark_submit.py:495} INFO - found org.apache.httpcomponents#httpclient;4.5.6 in central
[2024-11-24T10:08:31.222+0000] {spark_submit.py:495} INFO - found org.apache.httpcomponents#httpcore;4.4.10 in central
[2024-11-24T10:08:31.388+0000] {spark_submit.py:495} INFO - found commons-codec#commons-codec;1.11 in central
[2024-11-24T10:08:31.454+0000] {spark_submit.py:495} INFO - found commons-io#commons-io;2.5 in central
[2024-11-24T10:08:31.476+0000] {spark_submit.py:495} INFO - found commons-net#commons-net;3.6 in central
[2024-11-24T10:08:31.547+0000] {spark_submit.py:495} INFO - found commons-collections#commons-collections;3.2.2 in central
[2024-11-24T10:08:31.659+0000] {spark_submit.py:495} INFO - found org.eclipse.jetty#jetty-servlet;9.3.24.v20180605 in central
[2024-11-24T10:08:32.059+0000] {spark_submit.py:495} INFO - found org.eclipse.jetty#jetty-security;9.3.24.v20180605 in central
[2024-11-24T10:08:32.147+0000] {spark_submit.py:495} INFO - found org.eclipse.jetty#jetty-webapp;9.3.24.v20180605 in central
[2024-11-24T10:08:32.208+0000] {spark_submit.py:495} INFO - found org.eclipse.jetty#jetty-xml;9.3.24.v20180605 in central
[2024-11-24T10:08:32.298+0000] {spark_submit.py:495} INFO - found com.sun.jersey#jersey-servlet;1.19 in central
[2024-11-24T10:08:32.350+0000] {spark_submit.py:495} INFO - found log4j#log4j;1.2.17 in central
[2024-11-24T10:08:32.396+0000] {spark_submit.py:495} INFO - found commons-beanutils#commons-beanutils;1.9.3 in central
[2024-11-24T10:08:32.462+0000] {spark_submit.py:495} INFO - found org.apache.commons#commons-configuration2;2.1.1 in central
[2024-11-24T10:08:32.546+0000] {spark_submit.py:495} INFO - found org.apache.commons#commons-lang3;3.7 in central
[2024-11-24T10:08:32.596+0000] {spark_submit.py:495} INFO - found org.apache.commons#commons-text;1.4 in central
[2024-11-24T10:08:32.700+0000] {spark_submit.py:495} INFO - found org.apache.avro#avro;1.7.7 in central
[2024-11-24T10:08:32.941+0000] {spark_submit.py:495} INFO - found org.codehaus.jackson#jackson-core-asl;1.9.13 in central
[2024-11-24T10:08:33.018+0000] {spark_submit.py:495} INFO - found org.codehaus.jackson#jackson-mapper-asl;1.9.13 in central
[2024-11-24T10:08:33.102+0000] {spark_submit.py:495} INFO - found com.thoughtworks.paranamer#paranamer;2.3 in central
[2024-11-24T10:08:33.537+0000] {spark_submit.py:495} INFO - found org.apache.commons#commons-compress;1.18 in central
[2024-11-24T10:08:33.732+0000] {spark_submit.py:495} INFO - found com.google.re2j#re2j;1.1 in central
[2024-11-24T10:08:33.823+0000] {spark_submit.py:495} INFO - found com.google.protobuf#protobuf-java;2.5.0 in central
[2024-11-24T10:08:34.211+0000] {spark_submit.py:495} INFO - found com.google.code.gson#gson;2.2.4 in central
[2024-11-24T10:08:34.314+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-auth;3.2.1 in central
[2024-11-24T10:08:34.518+0000] {spark_submit.py:495} INFO - found com.nimbusds#nimbus-jose-jwt;4.41.1 in central
[2024-11-24T10:08:34.907+0000] {spark_submit.py:495} INFO - found com.github.stephenc.jcip#jcip-annotations;1.0-1 in central
[2024-11-24T10:08:35.705+0000] {spark_submit.py:495} INFO - found net.minidev#json-smart;2.3 in central
[2024-11-24T10:08:36.465+0000] {spark_submit.py:495} INFO - found net.minidev#accessors-smart;1.2 in central
[2024-11-24T10:08:36.601+0000] {spark_submit.py:495} INFO - found org.ow2.asm#asm;5.0.4 in central
[2024-11-24T10:08:36.652+0000] {spark_submit.py:495} INFO - found org.apache.curator#curator-framework;2.13.0 in central
[2024-11-24T10:08:36.707+0000] {spark_submit.py:495} INFO - found org.apache.curator#curator-client;2.13.0 in central
[2024-11-24T10:08:36.824+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-simplekdc;1.0.1 in central
[2024-11-24T10:08:36.911+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-client;1.0.1 in central
[2024-11-24T10:08:37.118+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerby-config;1.0.1 in central
[2024-11-24T10:08:37.444+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-core;1.0.1 in central
[2024-11-24T10:08:37.593+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerby-pkix;1.0.1 in central
[2024-11-24T10:08:37.735+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerby-asn1;1.0.1 in central
[2024-11-24T10:08:38.202+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerby-util;1.0.1 in central
[2024-11-24T10:08:38.399+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-common;1.0.1 in central
[2024-11-24T10:08:38.544+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-crypto;1.0.1 in central
[2024-11-24T10:08:38.765+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-util;1.0.1 in central
[2024-11-24T10:08:38.879+0000] {spark_submit.py:495} INFO - found org.apache.kerby#token-provider;1.0.1 in central
[2024-11-24T10:08:39.080+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-admin;1.0.1 in central
[2024-11-24T10:08:39.280+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-server;1.0.1 in central
[2024-11-24T10:08:39.542+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerb-identity;1.0.1 in central
[2024-11-24T10:08:39.892+0000] {spark_submit.py:495} INFO - found org.apache.kerby#kerby-xdr;1.0.1 in central
[2024-11-24T10:08:40.123+0000] {spark_submit.py:495} INFO - found org.apache.curator#curator-recipes;2.13.0 in central
[2024-11-24T10:08:40.195+0000] {spark_submit.py:495} INFO - found org.apache.htrace#htrace-core4;4.1.0-incubating in central
[2024-11-24T10:08:40.421+0000] {spark_submit.py:495} INFO - found com.fasterxml.jackson.core#jackson-databind;2.9.8 in central
[2024-11-24T10:08:40.553+0000] {spark_submit.py:495} INFO - found com.fasterxml.jackson.core#jackson-annotations;2.9.8 in central
[2024-11-24T10:08:40.783+0000] {spark_submit.py:495} INFO - found com.fasterxml.jackson.core#jackson-core;2.9.8 in central
[2024-11-24T10:08:40.979+0000] {spark_submit.py:495} INFO - found org.codehaus.woodstox#stax2-api;3.1.4 in central
[2024-11-24T10:08:41.064+0000] {spark_submit.py:495} INFO - found com.fasterxml.woodstox#woodstox-core;5.0.3 in central
[2024-11-24T10:08:41.299+0000] {spark_submit.py:495} INFO - found dnsjava#dnsjava;2.1.7 in central
[2024-11-24T10:08:41.547+0000] {spark_submit.py:495} INFO - found javax.servlet.jsp#jsp-api;2.1 in central
[2024-11-24T10:08:42.873+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-hdfs-client;3.2.1 in central
[2024-11-24T10:08:45.988+0000] {spark_submit.py:495} INFO - found com.squareup.okhttp#okhttp;2.7.5 in central
[2024-11-24T10:09:04.079+0000] {spark_submit.py:495} INFO - found com.squareup.okio#okio;1.6.0 in central
[2024-11-24T10:09:14.936+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-yarn-api;3.2.1 in central
[2024-11-24T10:09:20.988+0000] {spark_submit.py:495} INFO - found javax.xml.bind#jaxb-api;2.2.11 in central
[2024-11-24T10:09:21.546+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-yarn-client;3.2.1 in central
[2024-11-24T10:09:21.883+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-mapreduce-client-core;3.2.1 in central
[2024-11-24T10:09:22.605+0000] {spark_submit.py:495} INFO - found org.apache.hadoop#hadoop-yarn-common;3.2.1 in central
[2024-11-24T10:09:30.532+0000] {spark_submit.py:495} INFO - found javax.servlet#javax.servlet-api;3.1.0 in central
[2024-11-24T10:09:54.924+0000] {job.py:218} ERROR - Job heartbeat got an exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3371, in _wrap_pool_connect
    return fn()
           ^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 327, in connect
    return _ConnectionFairy._checkout(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 894, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 493, in checkout
    rec = pool._do_get()
          ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
    return self._create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 273, in _create_connection
    return _ConnectionRecord(self)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 388, in __init__
    self.__connect()
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 690, in __connect
    with util.safe_reraise():
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 686, in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/create.py", line 574, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 598, in connect
    return self.dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.OperationalError: could not translate host name "postgres" to address: Temporary failure in name resolution


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/jobs/job.py", line 213, in heartbeat
    heartbeat_callback(session)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/jobs/local_task_job_runner.py", line 261, in heartbeat_callback
    self.task_instance.refresh_from_db()
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 1821, in refresh_from_db
    _refresh_from_db(task_instance=self, session=session, lock_for_update=lock_for_update)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 510, in _refresh_from_db
    ti = TaskInstance.get_task_instance(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/api_internal/internal_api_call.py", line 115, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 1807, in get_task_instance
    return query.one_or_none()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2850, in one_or_none
    return self._iter().one_or_none()
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2916, in _iter
    result = self.session.execute(
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1716, in execute
    conn = self._connection_for_bind(bind)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1555, in _connection_for_bind
    return self._transaction._connection_for_bind(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 750, in _connection_for_bind
    conn = bind.connect()
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/future/engine.py", line 412, in connect
    return super(Engine, self).connect()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3325, in connect
    return self._connection_cls(self, close_with_result=close_with_result)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 96, in __init__
    else engine.raw_connection()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3404, in raw_connection
    return self._wrap_pool_connect(self.pool.connect, _connection)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3374, in _wrap_pool_connect
    Connection._handle_dbapi_exception_noconnection(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2208, in _handle_dbapi_exception_noconnection
    util.raise_(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3371, in _wrap_pool_connect
    return fn()
           ^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 327, in connect
    return _ConnectionFairy._checkout(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 894, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 493, in checkout
    rec = pool._do_get()
          ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
    return self._create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 273, in _create_connection
    return _ConnectionRecord(self)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 388, in __init__
    self.__connect()
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 690, in __connect
    with util.safe_reraise():
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 686, in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/create.py", line 574, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 598, in connect
    return self.dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "postgres" to address: Temporary failure in name resolution

(Background on this error at: https://sqlalche.me/e/14/e3q8)
[2024-11-24T10:09:57.610+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-24T10:11:13.909+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-24T10:12:18.600+0000] {spark_submit.py:495} INFO - found org.eclipse.jetty#jetty-util;9.3.24.v20180605 in central
[2024-11-24T10:12:18.892+0000] {spark_submit.py:495} INFO - found com.sun.jersey#jersey-core;1.19 in central
[2024-11-24T10:12:19.130+0000] {spark_submit.py:495} INFO - found javax.ws.rs#jsr311-api;1.1.1 in central
[2024-11-24T10:12:20.187+0000] {spark_submit.py:495} INFO - found com.sun.jersey#jersey-client;1.19 in central
[2024-11-24T10:12:28.280+0000] {spark_submit.py:495} INFO - found com.fasterxml.jackson.module#jackson-module-jaxb-annotations;2.9.8 in central
[2024-11-24T10:12:31.877+0000] {spark_submit.py:495} INFO - found com.fasterxml.jackson.jaxrs#jackson-jaxrs-json-provider;2.9.8 in central
[2024-11-24T10:12:45.707+0000] {spark_submit.py:495} INFO - found com.fasterxml.jackson.jaxrs#jackson-jaxrs-base;2.9.8 in central
[2024-11-24T10:14:49.926+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-24T10:24:14.328+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-24T10:27:21.219+0000] {local_task_job_runner.py:214} ERROR - Heartbeat time limit exceeded!
[2024-11-24T10:33:05.532+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-11-24T10:33:05.771+0000] {process_utils.py:132} INFO - Sending 15 to group 921. PIDs of all processes in the group: [922, 921]
[2024-11-24T10:33:05.776+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 921
[2024-11-24T10:33:05.837+0000] {taskinstance.py:2611} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-11-24T10:33:05.838+0000] {spark_submit.py:620} INFO - Sending kill signal to spark-submit
[2024-11-24T10:33:05.862+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-11-24T10:33:08.890+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 414, in submit
    self._process_spark_submit_log(iter(self._submit_sp.stdout))  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 463, in _process_spark_submit_log
    for line in itr:
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 2613, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-11-24T10:33:09.413+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=sensor_data_consumer, task_id=sensor_data_consumer, run_id=manual__2024-11-24T10:07:28.068559+00:00, execution_date=20241124T100728, start_date=20241124T100737, end_date=20241124T103309
[2024-11-24T10:33:11.057+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=922, status='terminated', started='10:07:39') (922) terminated with exit code None
[2024-11-24T10:33:11.062+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=921, status='terminated', exitcode=2, started='10:07:37') (921) terminated with exit code 2
